public/app/features/alerting/unified/components/rules/central-state-history/CentralAlertHistory.tsx

```diff
@@ -0,0 +1,368 @@
+            </Stack>
+          </Label>
+        }
+      >
+        <Input
+          id="eventsSearchInput"
+          prefix={<Icon name="search" />}
+          suffix={
+            showClearFilterSuffix && (
+              <Button fill="text" icon="times" size="sm" onClick={onClearFilterClick}>
+                <Trans i18nKey="central-alert-history.filter.button.clear"> Clear</Trans>
+              </Button>
+            )
+          }
+          placeholder={placeholder}
+          ref={ref}
+          {...rest}
+        />
+      </Field>
+    );
+  }
+);
+
+SearchFieldInput.displayName = 'SearchFieldInput';
+
+function EventRow({ record }: { record: LogRecord }) {
+  const styles = useStyles2(getStyles);
+  const [isCollapsed, setIsCollapsed] = useState(true);
+  return (
+    <div>
+      <div className={styles.header} data-testid="rule-group-header">
+        <CollapseToggle
+          size="sm"
+          className={styles.collapseToggle}
+          isCollapsed={isCollapsed}
+          onToggle={setIsCollapsed}
+        />
+        <Stack gap={0.5} direction={'row'} alignItems={'center'}>
+          <div className={styles.timeCol}>
+            <Timestamp time={record.timestamp} />
+          </div>
+          <div className={styles.transitionCol}>
+            <EventTransition previous={record.line.previous} current={record.line.current} />
+          </div>
+          <div className={styles.alertNameCol}>
+            {record.line.labels ? <AlertRuleName labels={record.line.labels} ruleUID={record.line.ruleUID} /> : null}
+          </div>
+          <div className={styles.labelsCol}>
+            <AlertLabels labels={record.line.labels ?? {}} limit={LIMIT_LABELS} />
+          </div>
+        </Stack>
+      </div>
+    </div>
+  );
+}
+
+function AlertRuleName({ labels, ruleUID }: { labels: Record<string, string>; ruleUID?: string }) {
+  const styles = useStyles2(getStyles);
+  const alertRuleName = labels['alertname'];
+  if (!ruleUID) {
+    return <Text>{alertRuleName}</Text>;
+  }
+  return (
+    <a href={`/alerting/${GRAFANA_DATASOURCE_NAME}/${ruleUID}/view`} className={styles.alertName}>
+      {alertRuleName}
+    </a>
+  );
+}
+
+interface EventTransitionProps {
+  previous: GrafanaAlertStateWithReason;
+  current: GrafanaAlertStateWithReason;
+}
+function EventTransition({ previous, current }: EventTransitionProps) {
+  return (
+    <Stack gap={0.5} direction={'row'}>
+      <EventState state={previous} />
+      <Icon name="arrow-right" size="lg" />
+      <EventState state={current} />
+    </Stack>
+  );
+}
+
+function EventState({ state }: { state: GrafanaAlertStateWithReason }) {
+  const styles = useStyles2(getStyles);
+
+  if (!isGrafanaAlertState(state) && !isAlertStateWithReason(state)) {
+    return (
+      <Tooltip content={'No recognized state'}>
+        <Icon name="exclamation-triangle" size="md" />
+      </Tooltip>
+    );
+  }
+  const stateWithReason = mapStateWithReasonToBaseState(state);
```

Comment:
nit: this is actually the `stateWithoutReason` :)

Perhaps to simplify things we should write a single function that extract both the "base" state and the "reason" from the alert state. We're currently using two different approaches to do that.

Probably something as simple as

```typescript
type State = GrafanaAlertState | PromAlertingRuleState

function parseState (state: string): [State, string | undefined] {
  const match = state.match(/(?<state>.*) \((?<reason>.*)\)/);
  if (!match?.groups?.state) {
    throw new Error('failed to parse state');
  }
  
  const parsedState = match.groups?.state as State;
  const parsedReason = match.groups?.reason;
  
  return [parsedState, parsedReason];
}
```

Reply:
As the `mapStateWithReasonToBaseState` was already implemented I found adding the new function for getting only the base state more reasonable. I will update the naming for the function and for the variable though.